ms2deepscore/models.py

```diff
@@ -0,0 +1,81 @@
+from typing import Tuple
+
+from tensorflow import keras
+
+
+class SiameseModel:
+    """
+    Class for training and evaluating a siamese neural network, implemented in Tensorflow Keras.
+    It consists of a dense 'base' network that produces an embedding for each of the 2 inputs. The
+    'head' model computes the cosine similarity between the embeddings.
```

Comment:
Nice! Having a .base and a head model is a very nice feature here (and actually solves my worries in #20 ).

Reply:
Cool!